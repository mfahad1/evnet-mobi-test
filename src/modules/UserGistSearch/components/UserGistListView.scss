.UserGistListView {
  background-color: white;
  width: 60vw;
  max-height: 70vh;
  overflow-y: auto;

  padding: 1rem;
  border-radius: 1rem;


  &::-webkit-scrollbar {
    width: 0.5rem;
    border-radius: 1rem;
  }


  &::-webkit-scrollbar-thumb {
    background-color: darkgrey;
    outline: 1px solid slategrey;
    border-radius: 1rem;
  }

  h1 {
    text-align: center;
  }
}

.UserGistListView_Item {
  border: 1px solid black;
  border-bottom: none;
  display: flex;
  padding: 1rem 0 1rem 1rem;
  flex-direction: column;

  a {
    color: black;
    text-decoration: none;
    padding-bottom: 1rem;
  }

  &:last-child {
    border-bottom: 1px solid black;
  }

  .badge {
    background-color: black;
    color: white;
    font-weight: bold;
    border-radius: 1rem;
    padding: 0.5rem;
    width: fit-content;
    font-size: 12px;
    margin-right: 0.5rem;
  }
}