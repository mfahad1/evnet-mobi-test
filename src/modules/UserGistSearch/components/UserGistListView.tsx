import { WithLoader } from '../../../components/WithLoader';
import { UserGistWithForkUsers } from '../../../hooks/services/useFetchUserGists';
import './UserGistListView.scss';

type UserGistListViewProps = {
  gists: UserGistWithForkUsers[] | undefined;
  loading: boolean;
  error: boolean;
};

export default function UserGistListView(props: UserGistListViewProps) {
  const { gists, loading, error } = props;

  return (
    <div className="UserGistListView">
      <WithLoader loading={loading} error={error}>
        <>
          <h1>Number of gist for searched user: {gists?.length || 0}</h1>
          {gists && gists.map((gist) => <UserGistListItem gist={gist} key={gist.id} />)}
        </>
      </WithLoader>
    </div>
  );
}

export function UserGistListItem({ gist }: { gist: UserGistWithForkUsers }) {
  const tags = [...new Set(Object.values(gist.files).map((file) => file.type))];

  return (
    <div className="UserGistListView_Item">
      <a href={gist.html_url} target="_blank" rel="noreferrer">
        {gist.description ? gist.description : gist.id}
      </a>
      <div className="AlignItemsCenter">
        {tags.length > 0 && <h3>TAGS:</h3>}
        {tags.map((tag) => (
          <span key={tag} className="badge">
            {tag}
          </span>
        ))}
      </div>

      <div className="UserGistListView_AvatarList AlignItemsCenter">
        {gist.forkedUsers.length > 0 && <h3>FORKED BY:</h3>}
        {gist.forkedUsers.map(({ html_url, avatar_url, login }) => (
          <a key={login} href={html_url} className="Avatar" target="_blank" rel="noreferrer">
            <img src={avatar_url} alt="avatar" />
          </a>
        ))}
      </div>
    </div>
  );
}
