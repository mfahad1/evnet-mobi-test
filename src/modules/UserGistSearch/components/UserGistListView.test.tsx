import * as faker from 'faker';
import renderer from 'react-test-renderer';
import UserGistListView, { UserGistListItem } from './UserGistListView';

export const fakerUser = () => ({
  id: faker.datatype.uuid(),
  description: faker.random.words(5),
  html_url: faker.datatype.string(5),
  files: { file: { type: 'application/json' } },
  forks_url: faker.datatype.string(5),
  login: faker.datatype.string(5),
  avatar_url: faker.datatype.string(5),
  forkedUsers: [
    {
      login: faker.datatype.string(5),
      avatar_url: faker.datatype.string(5),
      html_url: faker.datatype.string(5),
    },
  ],
});

const staticFakeUser = {
  id: 'testing',
  html_url: 'testing',
  files: { file: { type: 'application/json' } },
  forks_url: 'testing',
  forkedUsers: [
    {
      login: 'testing',
      avatar_url: 'testing',
      html_url: 'testing',
    },
  ],
};

describe('UserGistListItem', () => {
  it('should match UserGistListItem', () => {
    const elem = renderer.create(<UserGistListItem gist={staticFakeUser} />).toJSON();

    expect(elem).toMatchSnapshot();
  });

  it('should match UserGistListItem gist id', () => {
    const elem = renderer.create(<UserGistListItem gist={staticFakeUser} />).toJSON();

    expect(elem).toMatchSnapshot();
  });
});

describe('UserGistListView', () => {
  it('should match UserGistListView', () => {
    const elem = renderer.create(<UserGistListView gists={[staticFakeUser]} loading={false} error={false} />).toJSON();

    expect(elem).toMatchSnapshot();
  });
});
