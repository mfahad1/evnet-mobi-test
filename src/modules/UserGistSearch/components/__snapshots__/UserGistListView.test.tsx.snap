// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserGistListItem should match UserGistListItem 1`] = `
<div
  className="UserGistListView_Item"
>
  <a
    href="testing"
    rel="noreferrer"
    target="_blank"
  >
    testing
  </a>
  <div
    className="AlignItemsCenter"
  >
    <h3>
      TAGS:
    </h3>
    <span
      className="badge"
    >
      application/json
    </span>
  </div>
  <div
    className="UserGistListView_AvatarList AlignItemsCenter"
  >
    <h3>
      FORKED BY:
    </h3>
    <a
      className="Avatar"
      href="testing"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="avatar"
        src="testing"
      />
    </a>
  </div>
</div>
`;

exports[`UserGistListItem should match UserGistListItem gist id 1`] = `
<div
  className="UserGistListView_Item"
>
  <a
    href="testing"
    rel="noreferrer"
    target="_blank"
  >
    testing
  </a>
  <div
    className="AlignItemsCenter"
  >
    <h3>
      TAGS:
    </h3>
    <span
      className="badge"
    >
      application/json
    </span>
  </div>
  <div
    className="UserGistListView_AvatarList AlignItemsCenter"
  >
    <h3>
      FORKED BY:
    </h3>
    <a
      className="Avatar"
      href="testing"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="avatar"
        src="testing"
      />
    </a>
  </div>
</div>
`;

exports[`UserGistListView should match UserGistListView 1`] = `
<div
  className="UserGistListView"
>
  <h1>
    Number of gist for searched user: 
    1
  </h1>
  <div
    className="UserGistListView_Item"
  >
    <a
      href="testing"
      rel="noreferrer"
      target="_blank"
    >
      testing
    </a>
    <div
      className="AlignItemsCenter"
    >
      <h3>
        TAGS:
      </h3>
      <span
        className="badge"
      >
        application/json
      </span>
    </div>
    <div
      className="UserGistListView_AvatarList AlignItemsCenter"
    >
      <h3>
        FORKED BY:
      </h3>
      <a
        className="Avatar"
        href="testing"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="avatar"
          src="testing"
        />
      </a>
    </div>
  </div>
</div>
`;
